server/anp/io/feeding_services/anp_news_api.py

```diff
@@ -0,0 +1,169 @@
+
+                update.setdefault('private', {}).setdefault('sources', {})[source['id']] = {
+                    'title': source['title'],
+                    'last_item_id': item_details['id']
+                }
+
+                # parse item
+                parser = self.get_feed_parser(provider)
+                parsed_items.append(
+                    parser.parse(article=item_details, provider=provider)
+                )
+
+        return [parsed_items]
+
+    def _fetch_sources(self):
+        """
+        Fetch available sources and retrieves ids for `source_titles`
+
+        :return: a list of news sources.
+        """
+
+        response = self.get_url(
+            url=self.HTTP_SOURCES_URL
+        )
+        sources = json.loads(response.content)
+
+        if sources['hasError']:
+            msg = "Error in GET: '{}'. ErrorCode: '{}'. Description: '{}'".format(
+                self.HTTP_SOURCES_URL,
+                sources['data']['errorCode'],
+                sources['data']['description']
+            )
+            logger.error(msg)
+            raise IngestApiError.apiGeneralError(Exception(msg), self.provider)
+
+        return [
+            src for src in sources['data'] if src['title'].lower() in [
+                title.lower().strip() for title in [
+                    title_conf.lower().strip() for title_conf in self.config['source_titles'].split(',')
+                ]
+            ]
+        ]
+
+    def _fetch_items(self, source_id, to_item=None):
+        """
+        Fetch items ids for given source id
+        :param source_id:
+        :type provider: int
+        :return: a list of items ids.
+        """
+        payload = {}
+        if to_item:
+            payload['params'] = {'toItem': to_item}
+
+        response = self.get_url(
+            url=self.HTTP_ITEMS_URL.format(source_id=source_id), **payload
+        )
+        items = json.loads(response.content)
+
+        if items['hasError']:
+            msg = "Error in GET: '{}'. ErrorCode: '{}'. Description: '{}'".format(
+                self.HTTP_ITEMS_URL.format(source_id=source_id),
+                items['data']['errorCode'],
+                items['data']['description']
+            )
+            logger.error(msg)
+
+            return []
+        items['data']['items'].reverse()
+
+        return items['data']['items']
+
+    def _fetch_item_details(self, source_id, item_id):
+        """
+        Fetch item's details for given source id and item id
+        :param source_id:
+        :type provider: int
+        :param item_id:
+        :type provider: int
+        :return: a dict with item's details
+        """
+
+        response = self.get_url(
+            url=self.HTTP_ITEM_DETAILS_URL.format(source_id=source_id, item_id=item_id)
+        )
+        item_details = json.loads(response.content)
+
+        if item_details['hasError']:
```

Comment:
I think it can use some helper method which would do get_url and report error or return data, that code is duplicated here